provider/datagen/src/lib.rs

```diff
@@ -244,6 +244,8 @@ pub enum Out {
         /// Whether to gate each key on its crate name. This allows using the module
         /// even if some keys are not required and their dependencies are not included.
         insert_feature_gates: bool,
+        /// Whether to use separate crates to name types instead of the `icu` metacrate
+        use_separate_crates: bool,
```

Comment:
Praise: We can do this post-beta because it is non_exhaustive :)

Reply:
We could

Reply:
Wait no, I'm struct-constructing this, it doesn't seem to be non-exhaustive

Reply:
I would consider this an okay breaking change to make in a prerelease, fwiw.

Reply:
hmm, the enum is marked `#[non_exhaustive]`

Reply:
Apparently you have to mark each variant that is non-exhaustive as well. This is not something I want to do now though